api/v1beta2/cryostat_types.go

```diff
@@ -487,12 +488,38 @@ type TemplateConfigMap struct {
 
 // Authorization options provide additional configurations for the auth proxy.
 type AuthorizationOptions struct {
-	// Reference to a secret and file name containing the Basic authentication htpasswd file
+	// Configuration for OpenShift RBAC to define which OpenShift user accounts may access the Cryostat application.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	OpenShiftSsoRoles *OpenShiftSsoRoles `json:"openShiftSsoRoles,omitempty"`
+	// Reference to a secret and file name containing the Basic authentication htpasswd file. If deploying on OpenShift this
+	// defines additional user accounts that can access the Cryostat application, on top of the OpenShift user accounts which
+	// pass the OpenShiftSsoRoles checks. If not on OpenShift then this defines the only user accounts that have access.
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
 	BasicAuth *SecretFile `json:"basicAuth,omitempty"`
 }
 
+type OpenShiftSsoRoles struct {
+	// Disable OpenShift SSO integration and allow all users to access the application without authentication. This
+	// will also bypass the BasicAuth, if specified.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Disable OpenShift SSO",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	Disabled *bool `json:"disabled,omitempty"`
+	// The SubjectAccessReview that interactive clients (users visiting the application via web browser) must pass in
+	// order to access the application. If not specified, the default role required is "create pods/exec" in the Cryostat
+	// application's installation namespace.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	SubjectAccessReview *authzv1.ResourceAttributes `json:"subjectAccessReview,omitempty"`
+	// The TokenReview that programmatic clients (CLI utilities and others presenting Bearer auth tokens) must pass in
+	// order to access the application. If not specified, the default role required is "create pods/exec" in the Cryostat
+	// application's installation namespace.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	TokenReview *authzv1.ResourceAttributes `json:"tokenReview,omitempty"`
```

Comment:
Is TokenReview right here? That API is used to determine who a token belongs to isn't it? Also, do we need to have separate options for interactive vs non-interactive?

Reply:
I guess it is actually `TokenAccessReview`, not just `TokenReview`. I'll rename it.

> Also, do we need to have separate options for interactive vs non-interactive?

I think conceptually it makes sense to require the same access reviews either way, but unfortunately the way the proxy works is that you configure the interactive access as a list of objects that map a route prefix to a SAR, whereas the Bearer auth access is only a single object without any route prefix.

https://github.com/cryostatio/cryostat-helm/pull/132#discussion_r1571462608

So I think the most understandable model from the end user's point of view is probably to have a single `AccessReview` field, which is used for both TokenAccessReview and as a singleton list for SubjectAccessReview with the `/` prefix. The most flexible model is to have a prefixed list of SARs and a separate value for TAR.

Reply:
FWIW over in the Helm chart, I have it as a single Bearer TokenAccessReview and a list of SubjectAccessReviews.

Reply:
Ah so the delegate URLs option doesn't support a set of permissions per path? I agree with the simplest option then. One SubjectAccessReview field with one ResourceAttributes struct, and this will get used for both `openshift-sar` and `openshift-delegate-urls`.

Reply:
That's unfortunately correct, the delegate URLs option which enables Bearer token handling only takes a single permission per path prefix. But it also doesn't allow for inspecting the request at all beyond checking that the request path matches the path prefix, so we can't even differentiate between ex. `GET` and `POST` to apply different role mapping.